### PR TITLE
Add sim tests for multiple execution urls

### LINKED ISSUE #3845: backup eth1 node provider failover does not actually work?
**Describe the bug**
When a the primary eth1 node goes down and a second eth1 node begins serving requests, the logs get littered with messages as follows:
```
error: Error updating eth1 chain cache code=ETH1_ERROR_NON_CONSECUTIVE_LOGS, newIndex=123809, prevIndex=90472

Error: ETH1_ERROR_NON_CONSECUTIVE_LOGS
    at Eth1DepositsCache.add (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositsCache.ts:48:15)
    at Eth1DepositDataTracker.updateDepositCache (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositDataTracker.ts:174:5)
    at Eth1DepositDataTracker.update (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositDataTracker.ts:155:33)
    at Eth1DepositDataTracker.runAutoUpdate (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositDataTracker.ts:129:29)
```

**Expected behavior**
No errors (and no DB corruption?)

**Steps to Reproduce**
1. Have a fully synced node (unsure if this is required)
2. Specify two eth1 nodes with `--eth1.providerUrls`
3. Take the first of the two nodes offline and have it fall back (can be verified by seeing json rpc requests on the secondary)

**Screenshots**
n/a

**Desktop (please complete the following information):**
- OS: Ubuntu 20.04 LTS
- Version: `chainsafe/lodestar:v0.34.1` via docker
- Branch: n/a
- Commit hash: n/a

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Add a sim tests which checks the behavior of having multiple execution urls. 

**Description**

We want to be assure that if one execution node is not available, CL node should use the other available execution node. 

Closes #3845

**Steps to test or reproduce**

Run all sim tests.

### CODE REVIEW

File: packages/cli/test/sim/backup_eth_provider.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+env.nodes.push(newNode1);
+env.nodes.push(newNode2);
+
+await env.start({runTimeoutMs});
+await connectAllNodes(env.nodes);
+
+await waitForSlot(env.clock.getLastSlotOfEpoch(1), env.nodes, {silent: true, env});
+
+await newNode1.el.job.stop();
+await newNode2.el.job.stop();
```
Comment (dapplion) [inline packages/cli/test/sim/backup_eth_provider.test.ts]: What is this test testing exactly? A backup test should
- start CL 1 pointing to EL [1,2]
- ensure CL 1 progress ok targeting EL 1
- stop EL 1
- ensure CL 1 progresses ok targeting EL 2

Comment (nazarhussain) [reply packages/cli/test/sim/backup_eth_provider.test.ts]: This test is doing exactly the same. 

1. start CL/EL (1)
2. start CL/EL (2) with additional EL url pointing to EL1
3. start CL/EL (3) with additional EL url pointing to EL1
4. Wait for 1 epoch and assert nodes are progressing fine. 
5. Stop EL2
6. Stop EL3
7. Test that CL2 and CL3 are progressing fine and have same head as CL1. 

Do you think this approach needs to be changed?

Comment (dapplion) [reply packages/cli/test/sim/backup_eth_provider.test.ts]: Oh, then please comment the test to make it more obvious what going on. I did not understood that.

Comment (nazarhussain) [reply packages/cli/test/sim/backup_eth_provider.test.ts]: I update some code and added the comments. Also tried the nodes without additional engine URLs to make sure the test fails.

Comment (nazarhussain) [reply packages/cli/test/sim/backup_eth_provider.test.ts]: I made some changes and added the comments as well. Please re-review.

**Review by wemeetagain — APPROVED** [review]
lgtm

**Review by dapplion — APPROVED** [review]

**Pull request merged**